src/components/common/ColonyActions/ActionDetailsPage/DefaultMotion/MotionPhaseWidget/StakingWidget/StakingWidgetProvider.tsx

```diff
@@ -0,0 +1,262 @@
+    isObjection,
+    requiredStake,
+    rawStakes,
+  );
+
+  /*
+   * User's reputation in the domain is less than the amount still needed to stake, i.e. their
+   * staking ability is limited, not by their number of tokens activated, but by their (lack of) reputation.
+   */
+  const userNeedsMoreReputation =
+    userActivatedTokens.gt(maxUserStake) &&
+    maxUserStake.lt(remainingToStake.toString());
+
+  /*
+   * Occurs when the remaining amount to be staked is less than the user's minimum stake. (i.e. can't
+   * stake more than 100% of a motion)
+   */
+  const cantStakeMore =
+    remainingToStake.lte(minUserStake) && !remainingToStake.isZero();
+
+  const getErrorType = useCallback(
+    (limitExceeded: boolean) => {
+      if (!enoughReputation) {
+        return 'reputation';
+      }
+
+      if (!enoughTokens) {
+        return 'tokens';
+      }
+
+      if (cantStakeMore) {
+        return 'cantStakeMore';
+      }
+
+      if (userNeedsMoreReputation) {
+        return 'stakeMoreReputation';
+      }
+
+      if (limitExceeded) {
+        return 'stakeMoreTokens';
+      }
+
+      return null;
+    },
+    [enoughReputation, enoughTokens, userNeedsMoreReputation, cantStakeMore],
+  );
+
+  const canUserStakeNay = canUserStake && remainingToFullyNayStaked.gt(0);
+  const canBeStaked = canUserStake && remainingToStake.gt(0);
+
+  // todo: Add this data to the db once we handle the stake motion saga
+  const userStakes = {
+    yay: new Decimal(0),
+    nay: new Decimal(0),
+  };
+
+  const userStake = isObjection ? userStakes.nay : userStakes.yay;
+
+  const nativeTokenDecimals = colony?.nativeToken.decimals || 18;
+  const nativeTokenSymbol = colony?.nativeToken.symbol || '';
+
+  const stakingWidgetValues = useMemo(
+    () => ({
+      canBeStaked,
+      canUserStakeNay,
+      totalNAYStaked,
+      remainingToStake,
+      minUserStake,
+      maxUserStake,
+      yayPercentage: formatStakePercentage(yayPercentage),
+      nayPercentage: formatStakePercentage(nayPercentage),
+      totalPercentage,
+      enoughReputation,
+      enoughTokens,
+      isObjection,
+      setIsObjection,
+      userActivatedTokens,
+      isSummary,
+      setIsSummary,
+      motionId,
+      requiredStake,
+      userStake,
+      nativeTokenDecimals,
+      nativeTokenSymbol,
+      reputationLoading,
+      loadingStakeData,
+      getErrorType,
+    }),
+    [
+      canBeStaked,
+      canUserStakeNay,
+      remainingToStake,
+      totalNAYStaked,
+      minUserStake,
+      maxUserStake,
+      yayPercentage,
+      nayPercentage,
+      totalPercentage,
+      enoughReputation,
+      enoughTokens,
+      isObjection,
+      setIsObjection,
+      userActivatedTokens,
+      isSummary,
+      setIsSummary,
+      motionId,
+      requiredStake,
+      userStake,
+      nativeTokenDecimals,
+      nativeTokenSymbol,
+      reputationLoading,
+      loadingStakeData,
+      getErrorType,
+    ],
+  );
```

Comment:
Holly hell! There must be a better way. Can't we just pass the full motion data to this ? Because in it's current state it's just asking for trouble